grpc-server/subscribe_test.go

```diff
@@ -0,0 +1,114 @@
+package grpcserver
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/packethost/cacher/protos/cacher"
+	"github.com/packethost/pkg/log"
+	assert "github.com/stretchr/testify/require"
+	"github.com/tinkerbell/hegel/grpc/protos/hegel"
+	"github.com/tinkerbell/hegel/metrics"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestMain(m *testing.M) {
+	l, err := log.Init("github.com/tinkerbell/hegel")
+	if err != nil {
+		panic(nil)
+	}
+	metrics.Init(l)
+	x := m.Run()
+	l.Close() // not deferred because os.Exit would skip.
+	os.Exit(x)
+}
+
+func TestSubscribe(t *testing.T) {
+	t.Run("error", func(t *testing.T) {
+		terr := status.Error(codes.Unknown, "error pushing")
+		ctx, cancel, _, client := startServersAndConnectClient(t, nil, terr)
+		defer cancel()
+
+		w, err := client.Subscribe(ctx, &hegel.SubscribeRequest{ID: "doesn't matter"})
+		assert.NoError(t, err)
+		assert.NotNil(t, w)
+
+		hw, err := w.Recv()
+		assertGRPCError(t, err, terr)
+		assert.Nil(t, hw)
+	})
+
+	t.Run("good", func(t *testing.T) {
+		// id := uuid.Must(uuid.NewRandom()).String()
+		id := "bufconn"
+		data := map[string]string{}
+		value := fmt.Sprintf(`{"id": "%s", "ip": "%s"}`, id, id)
+		payload := fmt.Sprintf(`%s=%s`, id, value)
+		data[id] = value
```

Comment:
could be:
```suggestion
		value := fmt.Sprintf(`{"id": "%s", "ip": "%s"}`, id, id)
		payload := fmt.Sprintf(`%s=%s`, id, value)
		data := map[string]string{
			id: value,
		}
```
but not a big deal to change really

Reply:
Done